src/sql/workbench/contrib/executionPlan/browser/widgets/findExpensiveNodeWidget.ts

```diff
@@ -0,0 +1,310 @@
+	}
+
+	public focus() {
+		this._operationNameSelectBox.focus();
+	}
+
+	public getExpensiveOperationDelegate(): (cell: AzDataGraphCell) => number | undefined {
+		const getElapsedTimeInMs = (cell: AzDataGraphCell): number | undefined => cell.elapsedTimeInMs;
+		const getElapsedCpuTimeInMs = (cell: AzDataGraphCell): number | undefined => cell.elapsedCpuTimeInMs;
+		const getCost = (cell: AzDataGraphCell): number | undefined => cell.cost;
+		const getSubtreeCost = (cell: AzDataGraphCell): number | undefined => cell.subTreeCost;
+
+		const getRowsForAllExecutions = (cell: AzDataGraphCell): number | undefined => {
+			if (!cell.rowMetrics) {
+				return undefined;
+			}
+
+			let result = Number(cell.rowMetrics['actualRows']);
+			if (!result) {
+				result = Number(cell.rowMetrics['estimateRowsAllExecs']);
+			}
+
+			if (isNaN(result)) {
+				return undefined;
+			}
+
+			return result;
+		};
+
+		const getNumberOfRowsRead = (cell: AzDataGraphCell): number | undefined => {
+			if (!cell.rowMetrics) {
+				return undefined;
+			}
+
+			let result = Number(cell.rowMetrics['actualRowsRead']);
+			if (!result) {
+				result = Number(cell.rowMetrics['estimatedRowsRead']);
+			}
+
+			if (isNaN(result)) {
+				return undefined;
+			}
+
+			return result;
+		};
+
+		let expensiveOperationDelegate = getCost;
+		switch (this._selectedExpensiveOperationType) {
+			case ExpensiveOperationType.ActualElapsedTime:
+				expensiveOperationDelegate = getElapsedTimeInMs;
+				break;
+			case ExpensiveOperationType.ActualElapsedCpuTime:
+				expensiveOperationDelegate = getElapsedCpuTimeInMs;
+				break;
+			case ExpensiveOperationType.SubtreeCost:
+				expensiveOperationDelegate = getSubtreeCost;
+				break;
+			case ExpensiveOperationType.ActualNumberOfRowsForAllExecutions:
+				expensiveOperationDelegate = getRowsForAllExecutions;
+				break;
+			case ExpensiveOperationType.NumberOfRowsRead:
+				expensiveOperationDelegate = getNumberOfRowsRead;
+				break;
+		}
+
+		return expensiveOperationDelegate;
+	}
+}
+
+export class FindExpensiveOperationAction extends Action {
+	public static ID = 'qp.findExpensiveOperationAction';
+	public static LABEL = localize('findExpensiveOperationAction', 'Find (Enter)');
+
+	constructor() {
+		super(FindExpensiveOperationAction.ID, FindExpensiveOperationAction.LABEL, searchIconClassNames);
+	}
+
+	public override async run(context: FindExpensiveOperationWidget): Promise<void> {
+		const expensiveOperationDelegate: (cell: AzDataGraphCell) => number | undefined = context.getExpensiveOperationDelegate();
+
+		context.executionPlanDiagram.clearExpensiveOperatorHighlighting();
+		let result = context.executionPlanDiagram.highlightExpensiveOperator(expensiveOperationDelegate);
+		if (!result) {
+			context.notificationService.info(localize('invalidPropertyExecutionPlanMetric', 'Unable to locate a node using the specified metric.'));
+		}
+	}
+}
+
+export class TurnOffExpensiveOperationAction extends Action {
```

Comment:
please update the name to reflect the actual purpose: `TurnOffExpensiveHighlightingOperationAction`